### PR TITLE
Set z-value for outline boxes between events and track labels

### PR DESCRIPTION
**Pull request by Mahmood-Darwish**
This PR fixes an issue where if the user selects a thread state slice and leaves that slice selected but then drags a different thread over the slice, the outline drawn still shows above the thread that is being dragged.

Note this reverts https://github.com/google/orbit/pull/3941

Bug: http://b/239781929

### CODE REVIEW

File: src/OrbitGl/GlCanvas.cpp [hunk 1]
```diff
@@ -31,8 +31,8 @@ float GlCanvas::kZValueTrack = 0.01f;
 float GlCanvas::kZValueIncompleteDataOverlayPicking = 0.02f;
 float GlCanvas::kZValueEventBar = 0.03f;
 float GlCanvas::kZValueBox = 0.05f;
-float GlCanvas::kZValueBoxBorder = 0.06f;
-float GlCanvas::kZValueEvent = 0.07f;
+float GlCanvas::kZValueEvent = 0.06f;
+float GlCanvas::kZValueBoxBorder = 0.07f;
```
Comment (beckerhe) [inline src/OrbitGl/GlCanvas.cpp:35]: I'd like Pablo's opinion on this change. Could this affect other parts of the UI?

Or are these constants only specific to the thread dependencies feature?

Comment (Mahmood-Darwish) [reply src/OrbitGl/GlCanvas.cpp:35]: To me, it doesn't seem like these constants affected the UI when I ran the program. But I'm not sure if there are some small or very specific elements that have also changed. I did discuss this change with Pablo offline however and he okay-ed it.

Comment (beckerhe) [reply src/OrbitGl/GlCanvas.cpp:35]: Great. In that case he should definitely be the one who should review this. :-)

Comment (karupayun) [reply src/OrbitGl/GlCanvas.cpp:35]: I proposed initially BoxBorder in the original PR (it makes sense for me) and I completely missed https://github.com/google/orbit/pull/3941.

This switch was the correct solution, BoxBorder should be on top of Event. I guess that when we introduced it in the histogram we didn't consider Events at all.

**Review by karupayun — COMMENTED** [review]
Codewise LGTM, but I would ask for a better commit message. "fix outline issue" doesn't say anything to me. The title should be more like "Set z-value for outline boxes between events and track labels" 

We also need to mention in the body that we are reverting this PR https://github.com/google/orbit/pull/3941 and making another solution.

Also, a nit, the first letter should be Uppercase.

**Review by karupayun — APPROVED** [review]
LGTM! Thanks for addressing my comments!

**Review by karupayun — COMMENTED** [review]
Sorry, can I ask you to also modify in the same way the commit message, not only the PR?

**Comment by Mahmood-Darwish** [comment]
@karupayun I edited the commit message.

**Pull request merged**